src/core/positioning/nmeagnssreceiver.cpp

```diff
@@ -85,3 +105,61 @@ void NmeaGnssReceiver::handleStopLogging()
 {
   mLogFile.close();
 }
+
+void NmeaGnssReceiver::handleIMUCorrection( const QString &sentence )
+{
+  static const int PARAMETER_STATUS = 19;
+
+  static const int IMU_KQGEO_STATUS_OK = 1026;
+  static const int IMU_KQGEO_STATUS_OK_NEW = 1967106;
+
+  QStringList parameters = sentence.split( ',' );
+  if ( parameters.size() < PARAMETER_STATUS )
+    return;
+
+  // Parse Status
+  bool ok = false;
+  const int status = parameters[PARAMETER_STATUS].toInt( &ok );
+  if ( ok == false )
+  {
+    mIMUCorrectedPosition.valid = false;
+    return;
+  }
+
+  if ( status != IMU_KQGEO_STATUS_OK && status != IMU_KQGEO_STATUS_OK_NEW )
+  {
+    mIMUCorrectedPosition.valid = false;
+    return;
+  }
+
+  mIMUCorrectedPosition.valid = true;
+
+  // Parse other parameters
+
+  QDateTime utcDateTime = QDateTime::currentDateTime();
+  QTime time = QTime::fromString( parameters[1], "hhmmss.zzz" );
+  if ( time.isValid() )
+    utcDateTime.setTime( time );
+
+  mIMUCorrectedPosition.latitude = parameters[2].toDouble();
```

Comment:
Do we want to do validity check on double conversion here?

Reply:
Would be more error proof. What do you suggest to do if one of the parameters can't be parsed?

Reply:
I would say if we can't parse the latitude, longitude, or altitude, we mark the position status as invalid. I think we can be tolerant with other parameters.

Reply:
Ok. Another question, for the altitude I have seen there is some sort of correction for the Nmea position. Should the correction be applied to this altitude as well?

Reply:
Yeah. It'd be good to check with the happy guys what base altitude they use here

Reply:
They say the normal ellipsoid correction should be applied